src/cascadia/TerminalControl/TSFInputControl.cpp

```diff
@@ -175,52 +175,65 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
-        TextBlock().FontSize(fontSizePx);
+        // Make sure to unscale the font size to correct for DPI! XAML needs
+        // things in DIPs, and the fontSize is in pixels.
+        TextBlock().FontSize(fontSizePx / scaleFactor);
         TextBlock().FontFamily(Media::FontFamily(fontArgs->FontFace()));
 
-        const auto widthToTerminalEnd = _currentCanvasWidth - ::base::ClampedNumeric<double>(clientCursorPos.X);
+        const auto widthToTerminalEnd = _currentCanvasWidth - clientCursorInDips.x<double>();
         // Make sure that we're setting the MaxWidth to a positive number - a
         // negative number here will crash us in mysterious ways with a useless
         // stack trace
         const auto newMaxWidth = std::max<double>(0.0, widthToTerminalEnd);
         TextBlock().MaxWidth(newMaxWidth);
 
-        // Get window in screen coordinates, this is the entire window including tabs
-        const auto windowBounds = CoreWindow::GetForCurrentThread().Bounds();
+        // Get window in screen coordinates, this is the entire window including
+        // tabs. THIS IS IN DIPs
+        const auto windowBounds{ CoreWindow::GetForCurrentThread().Bounds() };
+        const til::point windowOrigin{ til::math::flooring, windowBounds };
 
-        // Convert from client coordinate to screen coordinate by adding window position
-        COORD screenCursorPos;
-        screenCursorPos.X = ::base::ClampAdd(clientCursorPos.X, ::base::ClampedNumeric<short>(windowBounds.X));
-        screenCursorPos.Y = ::base::ClampAdd(clientCursorPos.Y, ::base::ClampedNumeric<short>(windowBounds.Y));
+        // Get the offset (margin + tabs, etc..) of the control within the window
+        const til::point controlOrigin{ til::math::flooring,
+                                        this->TransformToVisual(nullptr).TransformPoint(Point(0, 0)) };
 
-        // get any offset (margin + tabs, etc..) of the control within the window
-        const auto offsetPoint = this->TransformToVisual(nullptr).TransformPoint(winrt::Windows::Foundation::Point(0, 0));
+        // The controlAbsoluteOrigin is the origin of the control relative to
+        // the origin of the displays. THIS IS IN DIPs
+        const til::point controlAbsoluteOrigin{ windowOrigin + controlOrigin };
 
-        // add the margin offsets if any
-        screenCursorPos.X = ::base::ClampAdd(screenCursorPos.X, ::base::ClampedNumeric<short>(offsetPoint.X));
-        screenCursorPos.Y = ::base::ClampAdd(screenCursorPos.Y, ::base::ClampedNumeric<short>(offsetPoint.Y));
+        // Convert the control origin to pixels
+        const til::point scaledFrameOrigin = controlAbsoluteOrigin * scaleFactor;
 
-        // Get scale factor for view
-        const double scaleFactor = DisplayInformation::GetForCurrentView().RawPixelsPerViewPixel();
-        const auto yOffset = ::base::ClampedNumeric<float>(_currentTextBlockHeight) - fontHeight;

```

Comment:
Looks like the `yOffset` was left out of the final calculation to textBounds.Y, this is used to adjust the position of the IME picker downwards when your composition hits the end of the Terminal and the textBlock starts to wrap and expand downwards like this pic in #5007: 
![image](https://user-images.githubusercontent.com/57155886/79780349-a0d07780-82f0-11ea-8ebf-af53b604f7c5.png)

Reply:
got it 
![image](https://user-images.githubusercontent.com/18356694/79800254-7ea31d80-8321-11ea-9675-9498b8fb17a4.png)